source/types/numeric.go

```diff
@@ -15,42 +15,23 @@
 package types
 
 import (
+	"math/big"
+
 	"github.com/jackc/pgx/v5/pgtype"
 )
 
 type NumericFormatter struct{}
 
-// Format coerces `pgtype.Numeric` to int or double depending on the exponent.
-// Returns error when value is invalid.
-func (NumericFormatter) Format(num pgtype.Numeric) (any, error) {
-	// N.B. The numeric type in pgx is represented by two ints.
-	//      When the type in Postgres is defined as `NUMERIC(10)' the scale is assumed to be 0.
-	//      However, pgx may represent the number as two ints e.g. 1200 -> (int=12,exp=2) = 12*10^2. as well
-	//      as a type with zero exponent, e.g. 121 -> (int=121,exp=0).
-	//      Thus, a Numeric type with positive or zero exponent is assumed to be an integer.
-	if num.Exp >= 0 {
-		i8v, err := num.Int64Value()
-		if err != nil {
-			return nil, err
-		}
-
-		v, err := i8v.Value()
-		if err != nil {
-			return nil, err
-		}
-
-		return v, nil
+// Format converts a pgtype.Numeric to a big.Rat.
+func (NumericFormatter) Format(num pgtype.Numeric) (*big.Rat, error) {
```

Comment:
Maybe we could rename this method to make it more explicit. e.g.: `BigRatFromNumeric` (or something like that)